unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/VideoPlayer/VideoPlayerHandler.cs

```diff
@@ -14,68 +14,120 @@ public class VideoPlayerHandler : IECSComponentHandler<PBVideoPlayer>
     {
         private static readonly string[] NO_STREAM_EXTENSIONS = new[] { ".mp4", ".ogg", ".mov", ".webm" };
 
+        private IParcelScene scene;
         private PBVideoPlayer lastModel = null;
         internal WebVideoPlayer videoPlayer;
 
+        // Flags to check if we can activate the video
+        internal bool isInsideScene = false;
+        internal bool isRendererActive = false;
+        internal bool hadUserInteraction = false;
+        internal bool isValidUrl = false;
+
         private readonly IInternalECSComponent<InternalVideoPlayer> videoPlayerInternalComponent;
+        private bool canVideoBePlayed => isInsideScene && isRendererActive && hadUserInteraction && isValidUrl;
 
         public VideoPlayerHandler(IInternalECSComponent<InternalVideoPlayer> videoPlayerInternalComponent)
         {
             this.videoPlayerInternalComponent = videoPlayerInternalComponent;
         }
 
-        public void OnComponentCreated(IParcelScene scene, IDCLEntity entity) { }
+        public void OnComponentCreated(IParcelScene scene, IDCLEntity entity)
+        {
+            this.scene = scene;
+            isRendererActive = CommonScriptableObjects.rendererState.Get();
+            isInsideScene = scene.isPersistent || scene.sceneData.sceneNumber == CommonScriptableObjects.sceneNumber.Get();
+
+            // We need to check if the user interacted with the application before playing the video,
+            // otherwise browsers won't play the video, ending up in a fake 'playing' state.
+            hadUserInteraction = Helpers.Utils.IsCursorLocked;
+
+            if (!hadUserInteraction)
+                Helpers.Utils.OnCursorLockChanged += OnCursorLockChanged;
+            if (!scene.isPersistent)
+                CommonScriptableObjects.sceneNumber.OnChange += OnSceneChanged;
+            CommonScriptableObjects.rendererState.OnChange += OnRendererStateChanged;
+        }
 
         public void OnComponentRemoved(IParcelScene scene, IDCLEntity entity)
         {
+            Helpers.Utils.OnCursorLockChanged -= OnCursorLockChanged;
+            CommonScriptableObjects.sceneNumber.OnChange -= OnSceneChanged;
+            CommonScriptableObjects.rendererState.OnChange -= OnRendererStateChanged;
```

Comment:
I don't think this `CommonScriptableObjects.rendererState` is working as it used to, so probably it must be changed for the loading screen event

Reply:
Done

Reply:
aparently `CommonScriptableObjects.isLoadingHUDOpen` is not being used anymore, I have to find an alternative...

Reply:
ok ended up using `DataStore_LoadingScreen`